pkg/apis/config/v1alpha3/types.go

```diff
@@ -17,38 +17,40 @@ limitations under the License.
 package v1alpha3
 
 import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	"sigs.k8s.io/kind/pkg/container/cri"
 )
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
 // Cluster contains kind cluster configuration
 type Cluster struct {
-	// TypeMeta representing the type of the object and its API schema version.
-	metav1.TypeMeta `json:",inline"`
+	TypeMeta `yaml:",inline" json:",inline"`
 
 	// Nodes contains the list of nodes defined in the `kind` Cluster
 	// If unset this will default to a single control-plane node
 	// Note that if more than one control plane is specified, an external
 	// control plane load balancer will be provisioned implicitly
-	Nodes []Node `json:"nodes,omitempty"`
+	Nodes []Node `yaml:"nodes,omitempty" json:"nodes,omitempty"`
```

Comment:
sorry for not having a look at this PR again.

i think this dropped support for user provided JSON kind config files?
which is not very critical but diverges from the "component config" / api-machinery trend.

Reply:
- this also has json tags? (reread the diff?)
- we never supported JSON config files. I don't intend to. yaml is a superset of json.

Reply:
Kubernetes needs a seperate JSON parser for other reasons (because it gives better errors?). yaml.v3 also gives better errors https://github.com/kubernetes/kubernetes/issues/78946

Reply:
missed the extra tags in the diff.